crates/viewer/re_viewport_blueprint/src/view_contents.rs

```diff
@@ -37,8 +40,17 @@ use crate::{ViewBlueprint, ViewProperty};
 pub struct ViewContents {
     pub blueprint_entity_path: EntityPath,
 
-    pub view_class_identifier: ViewClassIdentifier,
-    pub entity_path_filter: ResolvedEntityPathFilter,
+    view_class_identifier: ViewClassIdentifier,
+
+    /// Deserialized entity path filter.
+    ///
+    /// Consider this read-only.
+    entity_path_filter: ResolvedEntityPathFilter,
+
+    /// Update entity path filter.
+    ///
+    /// Mutations go to this value and should be saved to the blueprint store when they occur.
+    new_entity_path_filter: Arc<Mutex<ResolvedEntityPathFilter>>,
```

Comment:
This is such a common pattern in imgui. We should at some point think of coming up with a higher-level abstraction for it 🤔  (but not now)

Reply:
💯